File: FrontEnd/lib/domain/dtos/weather/weather_query_dto.dart
/// @file weather_query_dto.dart
/// @description DTO de consulta de clima. Se completará en Fase 2.
/// @module Weather
/// @layer Domain
library;
class WeatherQueryDto {
  const WeatherQueryDto({required this.location, this.days = 7});
  final String location;
  final int    days;
}

File: BackEnd/src/domain/dtos/weather/weather-query.dto.ts
/**
 * @file weather-query.dto.ts
 * @description DTO de consulta de clima. Se completará en Fase 2.
 * @module Weather
 * @layer Domain
 */
import { IsString, IsOptional, IsInt, Min, Max } from 'class-validator';

export class WeatherQueryDto {
  @IsString() location!: string;
  @IsOptional() @IsInt() @Min(1) @Max(14) days?: number;
}
